### DISCUSSION TITLE
License automaticaly detected but never automaticaly 'Concluded'

### DISCUSSION THREAD

**Discussion by YanickNoblanc** [fossology/fossology#2366] [Q&A]
Hello everyone,

I have been trying for several days to clear a decision on a license and enable 'Apply decision to all future occurrences of this file' so that all future uploads will be detected AND concluded as matching this license. Whatever I do, I can never manage to make it work as expected.
Here are the operations I have performed many times :

- From Fossology server, go to Admin/License admin/Select License
- Select the license I created and modify MYLICENSE 'License Text'
- Make sure Active=yes, Checked=yes, SPDX Compatible=yes, Detector type=Reference license, Conclusion=MYLICENSE, Reported license=MYLICENSE
- From a client machine scan (fossology REST API) a debian package containing a debian/copyright file containing the exacte same 'License Text'
- From fossology server, go to browse, select the package that has been uploaded/scanned
- With license browser, display file debian/copyright
- Select Action=+, for MYLICENSE (at this stage it is not very clear from documentation whether Action='+' or 'X' should be selected, but whatever is selected, next problems remain the same).
- Enable ' Apply decision to all future occurrences of this file'
- Set Clearing decision type=Identified
- Click on 'Submit'
- From a client machine scan (fossology REST API) a second Debian package containing same license

I expect this new package to have MYLICENSE automatically detected AND concluded. But it is never the case:
- MYLICENSE is systematically detected
- But MYLICENSE is almost never 'Concluded'

Among the many tests I did, I did see MYLICENSE automatically concluded only a couple of times, without any clue on the reason it worked.

I initially had a server running fossology 3.11.0 and after more than 20 scans not automatically concluding , I tried upgrading to fossology 4.2.1, and did again many scans but the issue remains the same. MYLICENSE is almost never automatically concluded.

If someone could have hints on what is going wrong, I would be gratefull.

**Comment by soulty** [comment]
您好，我是张笑宇，邮件已收到。

**Comment by YanickNoblanc** [comment]
Hi any info on the way to automaticaly detect and conclude a licence ?

**Comment by YanickNoblanc** [comment]
@soulty No I did not receive you email

**Comment by soulty** [comment]
您好，我是张笑宇，邮件已收到。

**Comment by YanickNoblanc** [comment]
Hello, I am still not able to automatically detect and conclude our proprietary licenses.
I unfortunately did not receive any mail from you concerning this issue (please mail to: yanick.noblanc@idemia.com).
I switched back to Fossology 3.11.0 (waiting for an official release to solve this other issue https://github.com/fossology/fossology/issues/1481).

Do you have any suggestions to overcome my Problem ?

**Reply by YanickNoblanc** [reply]
Looks like you have an au-response in place on this thread ....

**Reply by shaheemazmalmmd** [reply]
Hello @YanickNoblanc , 

Actually after marking the decision as global for license decisions. you also need to select "Apply global decisions for current upload" While uploading a new component. Please see the below image.

![screenShot77](https://user-images.githubusercontent.com/10155976/231385047-95f95b6a-7933-48e8-940a-5406782503f9.PNG)


You need to select the highlighted option to capture decisions from previous package.

**Comment by soulty** [comment]
您好，我是张笑宇，邮件已收到。

**Reply by YanickNoblanc** [reply]
Thank you very much. Is this option also possible when analysing a package using REST api, with url option as follows ?

curl  -k -s -S -L -X POST "${FOSSURL}/repo/api/v1/uploads" \
       -H "folderId: ${FOLDERID}" \
       -H 'uploadDescription: created by REST' \
       -H 'public: public' \
       -H 'Content-Type: application/json' \
       -H 'uploadType: url' \
       -H "Authorization: Bearer ${FOSSTOKEN}" \
       --data-raw "{
        \"url\": \"${PKGURL}\",
        \"name\": \"${PKGFILENAME}\"
        }" \
